File: Inventory/src/domain/model/Inventory.ts
import { v4 as uuidv4 } from 'uuid';

export class Inventory{
    public uuid: string;
    public name: string;
    public price: number;
    public stock: number;


    constructor(name: string, price: number, stock: number){
        this.uuid = uuidv4();
        this.name = name;
        this.price = price;
        this.stock = stock;
    }
}

File: Orders/orders/src/main/java/com/soa/orders/domain/models/Inventory.java
package com.soa.orders.domain.models;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class Inventory {
    private String uuid;
    private String name;
    private Integer price;
    private Long stock;

    public Inventory(String uuid, String name, Integer price, Long stock) {
        this.uuid = uuid;
        this.name = name;
        this.price = price;
        this.stock = stock;
    }
}
